Sources/Basic/OSLog.swift

```diff
@@ -0,0 +1,248 @@
+
+  #if canImport(os)
+    @usableFromInline let _type: os.OSSignpostType?
+
+    private init(_ _type: os.OSSignpostType?) {
+        self._type = _type
+    }
+  #else
+    private init() {}
+  #endif
+
+    /// Begins a signposted interval.
+    public static let begin: OSSignpostType = {
+      #if canImport(os)
+        if #available(OSX 10.14, *) {
+            return OSSignpostType(.begin)
+        } else {
+            return OSSignpostType(nil)
+        }
+      #else
+        return OSSignpostType()
+      #endif
+    }()
+
+    /// Ends a signposted interval.
+    public static let end: OSSignpostType = {
+      #if canImport(os)
+        if #available(OSX 10.14, *) {
+            return OSSignpostType(.end)
+        } else {
+            return OSSignpostType(nil)
+        }
+      #else
+        return OSSignpostType()
+      #endif
+    }()
+
+    /// Marks a point of interest in time with no duration.
+    public static let event: OSSignpostType = {
+      #if canImport(os)
+        if #available(OSX 10.14, *) {
+            return OSSignpostType(.event)
+        } else {
+            return OSSignpostType(nil)
+        }
+      #else
+        return OSSignpostType()
+      #endif
+    }()
+}
+
+/// An ID to disambiguate intervals in a signpost.
+///
+/// This is a thin wrapper for Darwin's signpost APIs to make them usable without
+/// platform and availability checks.
+public struct OSSignpostID {
+
+    private let __id: Any?
+
+  #if canImport(os)
+    @available(OSX 10.14, *)
+    @usableFromInline var _id: os.OSSignpostID {
+        return __id as! os.OSSignpostID
+    }
+  #endif
+
+    private init(_ __id: Any?) {
+        self.__id = __id
+    }
+
+    // Generates an ID guaranteed to be unique within the matching scope of the
+    // provided log handle.
+    public init(log: OSLog) {
+      #if canImport(os)
+        if #available(OSX 10.14, *) {
+            self.init(os.OSSignpostID(log: log._log))
+        } else {
+            self.init(nil)
+        }
+      #else
+        self.init(nil)
+      #endif
+    }
+
+    /// A convenience value for signpost intervals that will never occur
+    /// concurrently.
+    public static let exclusive: OSSignpostID = {
+      #if canImport(os)
+        if #available(OSX 10.14, *) {
+            return OSSignpostID(os.OSSignpostID.exclusive)
+        } else {
+            return OSSignpostID(nil)
+        }
+      #else
+        return OSSignpostID(nil)
+      #endif
+    }()
+}
+
+/// Emits a signpost.
+@inlinable public func os_signpost(
+    _ type: OSSignpostType,
+    log: OSLog = .default,
+    name: StaticString,
+    signpostID: OSSignpostID = .exclusive
+) {
+  #if canImport(os)
+    if #available(OSX 10.14, *) {
+        os.os_signpost(type: type._type!, log: log._log, name: name, signpostID: signpostID._id)
+    }
+  #endif
+}
+
+/// Emits a signpost.
+@inlinable public func os_signpost(
+    _ type: OSSignpostType,
+    log: OSLog = .default,
```

Comment:
I'm not sure if this is necessary with `@inlinable`, but shouldn't `#dsohandle` be passed through?

Reply:
Not sure, I need to ask someone